File: tests/cases/legacyGameTest.js
const webdriver = require('selenium-webdriver'),
    By = webdriver.By,
    until = webdriver.until;

let driver_fx = new webdriver.Builder()
    .forBrowser('firefox')
    .build();

let driver_chr = new webdriver.Builder()
    .forBrowser('chrome')
    .build();


greetingTest(driver_fx);
greetingTest(driver_chr);

function greetingTest(driver) {
  driver.get('http://localhost:5000/');

  driver.sleep(9000).then(function() {
    driver.findElement(By.xpath("//button[contains(.,'Set Parameters')]")).click();
  });

  driver.sleep(11000).then(function() {
    driver.findElement(By.xpath("//button[contains(.,'Start Game')]")).click();
  });

  driver.sleep(12000).then(function() {
    driver.findElement(By.id('startButton')).click();
  });

  driver.sleep(14000).then(function() {
    while (driver.getCurrentUrl() !== 'http://localhost:5000/user/results.html')
    {
        driver.sleep(600);
        driver.findElement(By.id('gameCanvas')).sendKeys(webdriver.Key.SPACE);
        console.log('Pressed Space');
    }

  });

  driver.sleep(19000).then(function() {
    driver.findElement(By.id('greeting')).getText().then(function(text) {
      if(text === 'Welcome, automatedtestingdemo!') {
        console.log('User Greeting Test passed');
      } else {
        console.log('User Greeting Test failed');
      }
      driver.quit();
    });
  });

}

File: tests/cases/legacyGameTest.py
import pyautogui;
import time;
import unittest 
from selenium import webdriver 
from selenium.webdriver.common.keys import Keys 

# inherit TestCase Class and create a new test class 
class legacyGameTest(unittest.TestCase): 

	# initialization of webdriver 
	def setUp(self): 
		self.driver = webdriver.Firefox() 

	# Test case method.
	def test_get_perfect_score(self): 
		
		# get driver 
		driver = self.driver 
		# get ractrainer web app using selenium 
		driver.get("https://ractrainer.web.app/") 

		time.sleep(2)

		# locate element using name 
		elem = driver.find_element_by_xpath("//button[contains(.,'Play Game')]") 
		# send data 
		elem.click() 

		time.sleep(2)

		# locate element using name 
		elem = driver.find_element_by_xpath("//button[contains(.,'Start Game')]") 
		# send data 
		elem.click() 

		time.sleep(2)

		# locate element using name 
		elem = driver.find_element_by_xpath("//button[contains(.,'START')]") 
		# send data 
		elem.click() 

		time.sleep(0.38)

		timeLimit = time.time() + 25
		while time.time() < timeLimit:
			pyautogui.keyDown('space')
			time.sleep(0.1)
			pyautogui.keyUp('space')
			time.sleep(0.3)

		# check score
		elem = driver.find_element_by_id("score").text
		assert "Score: 100%" in elem 

		time.sleep(1)

	# cleanup method called after every test performed 
	def tearDown(self): 
		self.driver.close() 

# execute the script 
if __name__ == "__main__": 
	unittest.main() 
